src/MongoDB.Driver/MongoClient.cs

```diff
@@ -338,13 +354,21 @@ private bool AreSessionsSupported(CancellationToken cancellationToken)
 
         private bool? AreSessionsSupported(ClusterDescription clusterDescription)
         {
-            if (clusterDescription.Servers.Any(s => s.IsDataBearing))
+            if (clusterDescription.LogicalSessionTimeout.HasValue)
             {
-                return clusterDescription.LogicalSessionTimeout.HasValue;
+                return true;
```

Comment:
This is a minor optimization.

If LogicalSessionTimeout has a value we know that sessions are supported without having to do any further analysis.

If it doesn't have a value we have to figure out (see else clause) whether it's because:
1. We don't know whether sessions are supported
2. Sessions aren't supported